app/src/main/java/de/westnordost/streetcomplete/quests/drinking_water_type/DrinkingWaterType.kt

```diff
@@ -0,0 +1,31 @@
+package de.westnordost.streetcomplete.quests.drinking_water_type
+
+import de.westnordost.streetcomplete.osm.Tags
+
+enum class DrinkingWaterType(val osmKey: String, val osmValue: String, val providesDrinkingWater: Boolean = true) {
+    WATER_FOUNTAIN_GENERIC("fountain", "drinking"),
+    WATER_FOUNTAIN_JET("fountain", "bubbler"),
+    WATER_FOUNTAIN_BOTTLE_REFILL_ONLY("fountain", "bottle_refill"),
+    WATER_TAP("man_made", "water_tap"),
+    HAND_PUMP("pump", "manual"),
+    WATER_WELL_WITHOUT_PUMP("pump", "no"),
+    SPRING("natural", "spring"),
+    DISUSED_DRINKING_WATER("disued:amenity", "drinking_water", false),
+}
+
+fun DrinkingWaterType.applyTo(tags: Tags) {
+    tags[this.osmKey] = this.osmValue
+    if (this == DrinkingWaterType.HAND_PUMP || this == DrinkingWaterType.WATER_WELL_WITHOUT_PUMP) {
+        tags["man_made"] = "water_well"
+    }
+    if (this.providesDrinkingWater) {
+        if (tags["disused:amenity"] == "drinking_water") {
+            tags.remove("disused:amenity")
+            tags["amenity"] = "drinking_water"
+        }
+    } else {
+        if (tags["amenity"] == "drinking_water") {
+            tags.remove("amenity")
+        }
+    }
+}
```

Comment:
Since the tagging logic is more complicated than key=value (more complicated than e.g. for `building=*`), it is more clear to put all of the logic into the `applyTo` function and let `DrinkingWaterType` be an enum with no further properties. I.e.

```kotlin
fun DrinkingWaterType.applyTo(tags: Tags) {
    when(this) {
        WATER_FOUNTAIN_GENERIC -> tags["fountain"] = "drinking"
        WATER_WELL_WITHOUT_PUMP -> {
            tags["man_made"] = "water_well"
            tags["pump"] = "no"
        }
        //...etc
    }
    if (this != DISUSED_DRINKING_WATER) {
        if (tags["disused:amenity"] == "drinking_water") {
            tags.remove("disused:amenity")
            tags["amenity"] = "drinking_water"
        }
    }
    //...etc
}

Reply:
done, testing it right now

this enum's parameters definitely has outgrown its usefulness